src/shared/hooks/reflect.ts

```diff
@@ -58,21 +80,31 @@ export function useReflect() {
 
     initReflect()
     updateUserInfo()
-  }, [reflectInitialized, name, realmName, stakingRealmColor, cursorTextColor])
+  }, [
+    reflect,
+    name,
+    realmName,
+    stakingRealmColor,
+    cursorTextColor,
+    initializeReflect,
+    reflectInitialized,
+  ])
```

Comment:
Looking at this dependency array I believe we are trying to achieve too many things in one useEffect and that will end up in a bigger mess. 

I suggest:
1. This shouldn't be happening in the hook - let's move it out of the hook and just initialize the `useState/useMemo` or whatever with that value if needed.
```
const reflectInstance =
        process.env.DISABLE_REFLECT === "true"
          ? null
          : new Reflect<ReflectMutators>({
              userID: nanoid(),
              roomID: "/",
              server: process.env.REFLECT_SERVER ?? "",
              mutators,
            })
```
2. When we need to do something like `useReflect(initializeReflect: boolean)` - pass the argument that tells the hook if it should be doing something from outside - that very often means we can move some logic out of this hook. So I think initialization and mutations should done by separate hooks.